src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -129,112 +183,570 @@ def __repr__(self):
+                % (FeatureFlagConstants.FROM, FeatureFlagConstants.TO, json.dumps(percentile_allocation, indent=2, ensure_ascii=False))
+            )
+
+        return cls(variant=variant, From=percentileFrom, to=perecentileTo)
+
+    def __repr__(self):
+        feature_percentile_allocation = {
+            FeatureFlagConstants.VARIANT: self.variant,
+            FeatureFlagConstants.FROM: self.From,
+            FeatureFlagConstants.TO: self.to,
+        }
+
+        return json.dumps(feature_percentile_allocation, indent=2, ensure_ascii=False)
+
+
+class FeatureAllocation:
+    """
+    Feature allocation class.
+
+    :ivar list FeatureUserAllocation user:
+        Determines how variants should be allocated for the feature to various users.
+    :ivar list FeatureGroupAllocation group:
+        Determines how variants should be allocated for the feature to various groups.
+    :ivar list FeaturePercentileAllocation percentile:
+        Determines how variants should be allocated for the feature to various users based on percentile.
+    """
+
+    def __init__(
+        self,
+        user=None,
+        group=None,
+        percentile=None,
+        default_when_enabled=None,
+        default_when_disabled=None,
+    ):
+        if user is not None:
+            self.user = user
+        if group is not None:
+            self.group = group
+        if percentile is not None:
+            self.percentile = percentile
+        if default_when_enabled is not None:
+            self.default_when_enabled = default_when_enabled
+        if default_when_disabled is not None:
+            self.default_when_disabled = default_when_disabled
+
+    @classmethod
+    def convert_from_json(cls, json_string):
+        """
+        Convert JSON string to FeatureAllocation object
 
         Args:
-            conditions_dict - Dictionary of {str, Any}
+            json_string - JSON string
 
         Return:
-            JSON serializable Dictionary
-    '''
+            FeatureAllocation object
+        """
+        allocation_json = json.loads(json_string)
+        default_when_disabled = allocation_json.get(
+            FeatureFlagConstants.DEFAULT_WHEN_DISABLED, None
+        )
+        default_when_enabled = allocation_json.get(
+            FeatureFlagConstants.DEFAULT_WHEN_ENABLED, None
+        )
+
+        allocation = cls(
+            default_when_enabled=default_when_enabled,
+            default_when_disabled=default_when_disabled,
+        )
+
+        allocation_user = allocation_json.get(FeatureFlagConstants.USER, None)
+
+        # Convert all users to FeatureUserAllocation object
+        if allocation_user:
+            allocation_user_list = []
+            for user in allocation_user:
+                feature_user_allocation = FeatureUserAllocation.convert_from_json(
+                    json.dumps(user, ensure_ascii=False)
+                )
+                if feature_user_allocation:
```

Comment:
can `convert_from_json` return `None`?

Reply:
No we check if the property is null first before calling the function

Reply:
So `if feature_user_allocation` should not be needed, right?

Reply:
Yes you are right its not needed.